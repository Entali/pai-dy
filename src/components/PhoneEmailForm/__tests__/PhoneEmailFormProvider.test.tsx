import {render} from '@testing-library/react'
import PhoneEmailFormProvider from '../PhoneEmailFormContext.tsx'
import Form from '../PhoneEmailForm.tsx'

const jestFn = jest.fn()

describe('PhoneEmailFormProvider', () => {
  it('should render w/ form', () => {
    const wrapper = render(
        <PhoneEmailFormProvider setMessage={jestFn} setModal={jestFn}>
          <Form setMessage={jestFn} setModal={jestFn}/>
        </PhoneEmailFormProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
