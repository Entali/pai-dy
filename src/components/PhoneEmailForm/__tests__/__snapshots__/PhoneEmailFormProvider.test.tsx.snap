// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneEmailFormProvider should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="phone-email-form"
        data-testid="phone-email-form"
      >
        <label
          class="phone-input"
        >
          <p
            class="text"
            style="font-size: 1rem; font-weight: 800; text-align: left;"
          >
            携帯電話番号
          </p>
          <div
            class="phone-input__input"
          >
            <span
              class="phone-input__country-code"
            >
              +81
            </span>
            <input
              autocomplete="tel"
              data-testid="phone-input__phone"
              placeholder="00-0000-0000"
              type="tel"
              value=""
            />
          </div>
        </label>
        <label
          class="email-input"
        >
          <p
            class="text"
            style="font-size: 1rem; font-weight: 800; text-align: left;"
          >
            メールアドレス
          </p>
          <input
            class="email-input__input"
            data-testid="email-input__email"
            placeholder="example@google.com"
            type="text"
            value=""
          />
        </label>
        <label
          class="checkbox"
        >
          <input
            type="checkbox"
          />
          <span
            class="checkbox__label"
          >
            次回から入力を省略
          </span>
        </label>
        <div
          class="phone-email-form__bottom"
        >
          <p
            class="text"
            style="font-size: 0.75rem; font-weight: 600; text-align: center;"
          >
            <a
              class="link"
              href="https://terms.paidy.com/personal-data"
              target="_blank"
            >
              利用規約・個人情報取扱条項
            </a>
            <span
              class="text"
              style="font-size: 0.75rem; font-weight: 400; text-align: left;"
            >
              に同意して
            </span>
          </p>
          <div
            class="button"
          >
            <input
              type="submit"
              value="次へ"
            />
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="phone-email-form"
      data-testid="phone-email-form"
    >
      <label
        class="phone-input"
      >
        <p
          class="text"
          style="font-size: 1rem; font-weight: 800; text-align: left;"
        >
          携帯電話番号
        </p>
        <div
          class="phone-input__input"
        >
          <span
            class="phone-input__country-code"
          >
            +81
          </span>
          <input
            autocomplete="tel"
            data-testid="phone-input__phone"
            placeholder="00-0000-0000"
            type="tel"
            value=""
          />
        </div>
      </label>
      <label
        class="email-input"
      >
        <p
          class="text"
          style="font-size: 1rem; font-weight: 800; text-align: left;"
        >
          メールアドレス
        </p>
        <input
          class="email-input__input"
          data-testid="email-input__email"
          placeholder="example@google.com"
          type="text"
          value=""
        />
      </label>
      <label
        class="checkbox"
      >
        <input
          type="checkbox"
        />
        <span
          class="checkbox__label"
        >
          次回から入力を省略
        </span>
      </label>
      <div
        class="phone-email-form__bottom"
      >
        <p
          class="text"
          style="font-size: 0.75rem; font-weight: 600; text-align: center;"
        >
          <a
            class="link"
            href="https://terms.paidy.com/personal-data"
            target="_blank"
          >
            利用規約・個人情報取扱条項
          </a>
          <span
            class="text"
            style="font-size: 0.75rem; font-weight: 400; text-align: left;"
          >
            に同意して
          </span>
        </p>
        <div
          class="button"
        >
          <input
            type="submit"
            value="次へ"
          />
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PhoneEmailFormProvider should render w/ form 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="phone-email-form"
        data-testid="phone-email-form"
      >
        <label
          class="phone-input"
        >
          <p
            class="text"
            style="font-size: 1rem; font-weight: 800; text-align: left;"
          >
            携帯電話番号
          </p>
          <div
            class="phone-input__input"
          >
            <span
              class="phone-input__country-code"
            >
              +81
            </span>
            <input
              autocomplete="tel"
              data-testid="phone-input__phone"
              placeholder="00-0000-0000"
              type="tel"
              value=""
            />
          </div>
        </label>
        <label
          class="email-input"
        >
          <p
            class="text"
            style="font-size: 1rem; font-weight: 800; text-align: left;"
          >
            メールアドレス
          </p>
          <input
            class="email-input__input"
            data-testid="email-input__email"
            placeholder="example@google.com"
            type="text"
            value=""
          />
        </label>
        <label
          class="checkbox"
        >
          <input
            type="checkbox"
          />
          <span
            class="checkbox__label"
          >
            次回から入力を省略
          </span>
        </label>
        <div
          class="phone-email-form__bottom"
        >
          <p
            class="text"
            style="font-size: 0.75rem; font-weight: 600; text-align: center;"
          >
            <a
              class="link"
              href="https://terms.paidy.com/personal-data"
              target="_blank"
            >
              利用規約・個人情報取扱条項
            </a>
            <span
              class="text"
              style="font-size: 0.75rem; font-weight: 400; text-align: left;"
            >
              に同意して
            </span>
          </p>
          <div
            class="button"
          >
            <input
              type="submit"
              value="次へ"
            />
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="phone-email-form"
      data-testid="phone-email-form"
    >
      <label
        class="phone-input"
      >
        <p
          class="text"
          style="font-size: 1rem; font-weight: 800; text-align: left;"
        >
          携帯電話番号
        </p>
        <div
          class="phone-input__input"
        >
          <span
            class="phone-input__country-code"
          >
            +81
          </span>
          <input
            autocomplete="tel"
            data-testid="phone-input__phone"
            placeholder="00-0000-0000"
            type="tel"
            value=""
          />
        </div>
      </label>
      <label
        class="email-input"
      >
        <p
          class="text"
          style="font-size: 1rem; font-weight: 800; text-align: left;"
        >
          メールアドレス
        </p>
        <input
          class="email-input__input"
          data-testid="email-input__email"
          placeholder="example@google.com"
          type="text"
          value=""
        />
      </label>
      <label
        class="checkbox"
      >
        <input
          type="checkbox"
        />
        <span
          class="checkbox__label"
        >
          次回から入力を省略
        </span>
      </label>
      <div
        class="phone-email-form__bottom"
      >
        <p
          class="text"
          style="font-size: 0.75rem; font-weight: 600; text-align: center;"
        >
          <a
            class="link"
            href="https://terms.paidy.com/personal-data"
            target="_blank"
          >
            利用規約・個人情報取扱条項
          </a>
          <span
            class="text"
            style="font-size: 0.75rem; font-weight: 400; text-align: left;"
          >
            に同意して
          </span>
        </p>
        <div
          class="button"
        >
          <input
            type="submit"
            value="次へ"
          />
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
